client/site/main.js

```diff
@@ -33,33 +91,34 @@ function spawnModules(canvas, memory) {
     workers.push(spawnModule('graphics', memory, canvas));
 }
 
+function throwMissingOffscreenCanvasSupport() {
+    document.write('your browser does not seem to support OffscreenCanvas.');
+}
+
 function createCanvas() {
     let canvas = document.getElementById('c');
-    canvas = canvas.transferControlToOffscreen();
-    return canvas;
+    if (canvas.transferControlToOffscreen === undefined)
```

Comment:
IIRC `typeof <expr> === 'undefined'` is the safest way to check if a value is declared.

Reply:
Since `transferControlToOffscreen` must be either `undefined` or a function, this seems unnecessary for me. I am no expert to js, so if `typeof <> === 'undefined'` is a very common pattern, I will change this.

Reply:
Oh, wasn't it standard, to check for existence of a function, by just doing: `if (func)`?

Reply:
So, I just did some more research: assuming that `canvas` exists and is an object, it's generally safe to check for a property's existence by ensuring that it's not `undefined`.

However, you can't check for a variable by comparing it to `undefined`, in *that* case you need to use `typeof`. If you want to ensure though that `canvas.transferControlToOffscreen` is a function you may want to do something like `typeof val === 'function'` :)

Reply:
Oof, thanks for your research. Yes, I'm gonna fix this